xml/encoder_test.go

```diff
@@ -0,0 +1,447 @@
+package xml_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/awslabs/smithy-go/xml"
+)
+
+func TestEncoder(t *testing.T) {
+	encoder := xml.NewEncoder()
+
+	obj := encoder.RootElement()
+	obj.Key("stringKey").String("stringValue")
+	obj.Key("integerKey").Integer(1024)
+	obj.Key("floatKey").Float(3.14)
```

Comment:
What is the root node in this tests? XML requires one root tag to enclose the entire document. This test is actually writing three different XML documents.